src/stream/src/executor/top_n/top_n_cache.rs

```diff
@@ -861,3 +807,86 @@ impl AppendOnlyTopNCacheTrait for TopNCache<true> {
         Ok(())
     }
 }
+
+/// Used to manage staging changes for one `TopNCache` (for one group).
+/// It should be maintained when an entry is inserted or deleted from the `middle` cache.
+#[derive(Debug, Default)]
+pub struct TopNStaging {
```

Comment:
Is it possible to reuse `StreamChunkCompactor`?

Reply:
Since we can achieve it with ease, I want to avoid deserialize those rows and construct `StreamChunk` with wastes.

Reply:
> avoid deserialize those rows and construct StreamChunk with wastes

This sounds reasonable. Does `RowOpMap` work?

> with ease

Haven't read the code, but you also made a mistake. So I'm thinking it's not trivial 😄

Reply:
It's pretty straightforward actually. I made a mistake because I'm stupid😄